src/new/wallet.js

```diff
@@ -195,22 +195,118 @@ class HathorWallet extends EventEmitter {
     }
   }
 
-  getAllAddresses() {
+  /**
+   * Old getAllAddresses method used to keep compatibility
+   * with some methods that used to need it
+   *
+   * @return {Array} Array of addresses (string)
+   *
+   * @memberof HathorWallet
+   * @inner
+   **/
+  getAllAddressesSimple() {
     storage.setStore(this.store);
     return wallet.getAllAddresses();
   }
 
+  /**
+   * Return all addresses of the wallet with info of each of them
+   *
+   * @return {Promise<Array>} Array of objects { address, index, transactions } where transactions is the count of txs for this address
+   *
+   * @memberof HathorWallet
+   * @inner
+   **/
+  async getAllAddresses() {
```

Comment:
The idea is to have both methods (from both facades) working the same way. The one from wallet service just returns the array of addresses. I could change the other one as well but not sure if we should, the wallet service API already returns the full array anyway

Reply:
I feel that an iterator would make the use simpler because we might need to paginate large returns, and we wouldn't have to request all pages before returning. Besides that, it might use less memory in cases with a large number of addresses.

Reply:
I think we should do it now because it seems quick to be develop and it would barely affect the current code. It just keeps things more flexible for future features.

Reply:
I agree. Done here https://github.com/HathorNetwork/hathor-wallet-lib/pull/221/commits/61783cc94fbfb8328609a96417c0fffa6af8da2c

Reply:
That's great! Thanks!